projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.ts

```diff
@@ -105,4 +110,28 @@ export class ResourceHolderComponent implements OnInit{
     return item;
   }
 
+  updateQueryParams() {
+    const queryParams = {
+      page: this.pagination.currentPage + 1,
+      limit: this.pagination.pageSize,
+      type: this.filters.current.type || "",
+      search: btoa(this.filters.search) || '',
+      sort_by: this.sortOptions.sort_by || '',
+      sort_order: this.sortOptions.sort_order || ''
+    };
+    this.commonService.updateQueryParams(queryParams);
+  }
+
+  getQueryParams() {
```

Comment:
already function is there to get params right ? then why another function here ?